src/gui/qgsmetadatawidget.h

```diff
@@ -0,0 +1,245 @@
+     * Remove a selected right.
+     */
+    void removeRight() const;
+
+    /**
+     * Add a new constraint.
+     */
+    void addConstraint() const;
+
+    /**
+     * Remove a constraint.
+     */
+    void removeConstraint() const;
+
+    /**
+     * Fill the CRS field from the layer.
+     */
+    void fillCrsFromLayer() const;
+
+    /**
+     * Add a new contact.
+     */
+    void addContact() const;
+
+    /**
+     * Remove a selected contact.
+     */
+    void removeContact() const;
+
+    /**
+     * Update the contact details according to the selection in the contact list.
+     */
+    void updateContactDetails() const;
+
+    /**
+     * Add a new link.
+     */
+    void addLink() const;
+
+    /**
+     * Remove a selected link.
+     */
+    void removeLink() const;
+
+    /**
+     * Add a new history.
+     */
+    void addHistory();
+
+    /**
+     * Remove a selected history.
+     */
+    void removeHistory() const;
+
+    /**
+     * Function to fill combobox like language, type.
+     */
+    void fillComboBox() const;
+
+    /**
+     * Fill the wizard from values in the layer metadata object.
+     */
+    void setPropertiesFromLayer() const;
+
+    /**
+     * Save all fields in a QgsLayerMetadata object.
+     */
+    void saveMetadata( QgsLayerMetadata &layerMetadata ) const;
+
+    /**
+     * Check if values in the wizard are correct.
+     */
+    bool checkMetadata() const;
+
+    /**
+     * Returns a list of languages available by default in the wizard.
+     */
+    static QMap<QString, QString> parseLanguages();
+
+    /**
+     * Returns a list of licences available by default in the wizard.
+     */
+    static QStringList parseLicenses();
+
+    /**
+     * Returns a list of link types available by default in the wizard.
+     * \see https://github.com/OSGeo/Cat-Interop/blob/master/LinkPropertyLookupTable.csv
+     */
+    static QStringList parseLinkTypes();
+
+    /**
+     * Returns a list of MIME types available by default in the wizard.
+     * \see https://fr.wikipedia.org/wiki/Type_MIME
+     */
+    static QStringList parseMimeTypes();
+
+    /**
+     * Returns a list of types available by default in the wizard.
+     */
+    static QMap<QString, QString> parseTypes();
+
+    /**
+     * Saves the metadata to the layer.
+     */
+    void acceptMetadata();
+
+  private:
+    void updatePanel() const;
+    void addDefaultCategory() const;
+    void addNewCategory();
+    void removeCategory() const;
+    QStringList mDefaultCategories;
+    QgsMapLayer *mLayer = nullptr;
+    QgsLayerMetadata mMetadata;
+    QStandardItemModel *mConstraintsModel = nullptr;
+    QStandardItemModel *mLinksModel = nullptr;
+    QStringListModel *mCategoriesModel = nullptr;
+    QStringListModel *mDefaultCategoriesModel = nullptr;
+    QStringListModel *mRightsModel = nullptr;
+    QStringListModel *mHistoryModel = nullptr;
+    void syncFromCategoriesTabToKeywordsTab() const;
+};
+
+#ifndef SIP_RUN
+
+/**
```

Comment:
Also put a `\\\@cond PRIVATE` here, and `\\\@endcond` at the end of the file

This hides the classes from the doxygen files, making them truely private and not part of stable api